pkg/gce-pd-csi-driver/controller.go

```diff
@@ -515,8 +515,34 @@ func (gceCS *GCEControllerServer) ValidateVolumeCapabilities(ctx context.Context
 
 func (gceCS *GCEControllerServer) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (*csi.ListVolumesResponse, error) {
 	// https://cloud.google.com/compute/docs/reference/beta/disks/list
-	// List volumes in the whole region? In only the zone that this controller is running?
-	return nil, status.Error(codes.Unimplemented, "")
+	if req.MaxEntries > 500 || req.MaxEntries < 0 {
```

Comment:
Or instead of failing, if the caller passes in > 500, can we just only return 500 and set NextToken?

Maybe we should add a clarification in the CSI spec, that plugin should be able to return fewer than MaxEntries + NextToken if they cannot support the requested?

Reply:
ah this was a brain fart. it is *MAX*entries after-all. So technically returning less than that and paginating is fine :)